br/pkg/utils/register.go

```diff
@@ -108,9 +132,36 @@ func (tr *TaskRegister) grant(ctx context.Context) (*clientv3.LeaseGrantResponse
 	return lease, nil
 }
 
-// RegisterTask firstly put its key to PD with a lease,
-// and start to keepalive the lease in the background.
-func (tr *TaskRegister) RegisterTask(c context.Context) error {
+// RegisterTaskOnce implements the TaskRegister interface
+func (tr *taskRegister) RegisterTaskOnce(ctx context.Context) error {
+	resp, err := tr.client.Get(ctx, tr.key)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if len(resp.Kvs) == 0 {
+		lease, err2 := tr.grant(ctx)
+		if err2 != nil {
+			return errors.Annotatef(err2, "failed grant a lease")
+		}
+		tr.curLeaseID = lease.ID
+		_, err2 = tr.client.KV.Put(ctx, tr.key, "", clientv3.WithLease(lease.ID))
+		if err2 != nil {
+			return errors.Trace(err2)
+		}
+	} else {
+		// if the task is run distributively, like IMPORT INTO, we should refresh the lease ID,
+		// in case the owner changed during the registration, and the new owner create the key.
+		tr.curLeaseID = clientv3.LeaseID(resp.Kvs[0].Lease)
```

Comment:
Does it want to prevent that, the new leader overwrites the key, but some peers are still keepalive for the overwritten key?

Reply:
yes, other tidb might keeps a stale lease id